src/vm.rs

```diff
@@ -136,13 +155,42 @@ impl<'a, E: UserDefinedError> EbpfVm<'a, E> {
     ///
     /// // Instantiate a VM.
     /// let executable = EbpfVm::<UserError>::create_executable_from_text_bytes(prog, None).unwrap();
-    /// let mut vm = EbpfVm::<UserError>::new(executable.as_ref()).unwrap();
+    /// let mut vm = EbpfVm::<UserError>::new(executable.as_ref(), &[], &[]).unwrap();
     /// ```
-    pub fn new(executable: &'a dyn Executable<E>) -> Result<EbpfVm<'a, E>, EbpfError<E>> {
+    pub fn new(
+        executable: &'a dyn Executable<E>,
+        mem: &[u8],
+        granted_regions: &[MemoryRegion],
+    ) -> Result<EbpfVm<'a, E>, EbpfError<E>> {
+        let (prog_addr, prog) = executable.get_text_bytes()?;
+        let frames = CallFrames::default();
+        let mut memory_mapping = MemoryMapping::new();
+        memory_mapping.add_regions(granted_regions.to_vec());
+        for ptr in frames.get_stacks() {
```

Comment:
Is "ptr" here a hold over from the naming in `get_stacks`?  Since `get_stacks` returns a vec of regions, can we use `memory_mapping.extend` here?